glean-core/android/src/main/java/mozilla/telemetry/glean/private/TimingDistributionMetricType.kt

```diff
@@ -144,6 +142,30 @@ class TimingDistributionMetricType internal constructor(
         }
     }
 
+    override fun accumulateSamples(samples: LongArray) {
+        if (!shouldRecord()) {
```

Comment:
There's one tricky part here: in glean-ac, if any sample is negative, we report an error. However, doesn't look like we allow for error reporting in Kotlin-land.

Reply:
Could we handle it in the FFI layer by checking the "sign bit" on the u64 coming in?  Something like `if value & 0x8000000000000000` ... report error...?

Reply:
We can just take a `i64` properly instead, matching `Long` from the Kotlin side, then do any underflow checks

Reply:
All right, I'll go for the `i64` route!